### DISCUSSION TITLE
Error occurred prerendering page and Export encountered errors on following paths (Next.js)

### DISCUSSION THREAD

**Discussion by humailsaleem123** [vercel/next.js#44928] [Help]
The project was working fine in development mode but these two errors occur when I export as production npm run build. I use getStaticProps and getStaticPath fetch from an API route.
I was also trying to do fetch API in another function and then call in getStaticProps but still getting this issue.Can anybody Please help me :)) , Thanks In Advance <3

 
when I type npm run build this errors occur:

    Error occurred prerendering page "/books/coding/63b6a7f947e0f6bade2be7eb". Read more: https://nextjs.org/docs/messages/prerender-error
        TypeError: fetch failed
            at Object.fetch (node:internal/deps/undici/undici:11118:11)
            at async loadDynamicData (E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\.next\server\pages\books\[book_category]\[_id].js:497:17)
            at async getStaticProps (E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\.next\server\pages\books\[book_category]\[_id].js:486:18)
            at async renderToHTML (E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\node_modules\next\dist\server\render.js:385:20)
            at async E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\node_modules\next\dist\export\worker.js:277:36
            at async Span.traceAsyncFn (E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\node_modules\next\dist\trace\trace.js:79:20)
        (node:4436) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
        (Use `node --trace-warnings ...` to show where the warning was created)
        info  - Generating static pages (26/26)
    
        > Build error occurred
        Error: Export encountered errors on following paths:
                /books/[book_category]/[_id]: /books/coding/63b6a84147e0f6bade2be7ec
            at E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\node_modules\next\dist\export\index.js:409:19
            at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
            at async Span.traceAsyncFn (E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\node_modules\next\dist\trace\trace.js:79:20)
            at async E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\node_modules\next\dist\build\index.js:1345:21
            at async Span.traceAsyncFn (E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\node_modules\next\dist\trace\trace.js:79:20)
            at async E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\node_modules\next\dist\build\index.js:1205:17
            at async Span.traceAsyncFn (E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\node_modules\next\dist\trace\trace.js:79:20)
            at async Object.build [as default] (E:\HUMAIL\NEXT.JS FOLDER\Books-Stack Project\Frontend\node_modules\next\dist\build\index.js:66:29)

books/category

**[Category].tsx**

    import React from "react";
    import BooksPage from "../../../Components/Books/BooksPage";
    
    export async function getStaticPaths() {
    
      const res = await fetch('http://localhost:4000/booksApi')
      const data = await res.json();
      console.log(data);
    
      const paths = data.map((item: any) => ({
        params: {
          book_category: item.book_category.toString(),
        },
      }));
    
      return { paths, fallback: false }
    }
    
    export const getStaticProps = async (context: any) => {
    
      const category = context.params.book_category;
    
      const data = await loadDynamicData(category);
    
      return {
        props: {
          data,
        },
      };
    };
    
    export async function loadDynamicData(category: any) {
    
      const res = await fetch(`http://localhost:4000/booksApi/${category}`);
      const data = await res.json();
    
      return data;
    }
    
    function DynamicCategory({ data }: any) {
    
      return (
    
        <div>
    
          <BooksPage getData={data} />
        </div>
      );
    }
    
    export default DynamicCategory;

books/category/id

**[id].tsx**

    import React from 'react'
    import ProductDetail from "../../../../Components/ProductDetail/ProductDetail";
    
    export async function getStaticPaths() {
    
      const res = await fetch("http://localhost:4000/booksApi");
      const data = await res.json();
    
      const paths = data.map((item: any) => ({
    
        params: {
          _id: item._id.toString(),
          book_category: item.book_category.toString()
    
        }
    
      }));
    
      return {
    
        paths,
        fallback: false
      }
    
    }
    
    export const getStaticProps = async (context: any) => {
    
      const id = context.params._id;
    
      const data = await loadDynamicData(id);
    
      return {
        props: {
          data,
        },
      };
    };
    
    export async function loadDynamicData(id: any) {
    
      const res = await fetch(`http://localhost:4000/booksApi/id/${id}`);
      const data = await res.json();
    
      return data;
    
    }
    
    function DynamicID({ data }: any) {
      return (
        <ProductDetail data={data} />
      )
    }
    export default DynamicID

**Comment by icyJoseph** [comment]
Well the first question is, who's running `http://localhost/booksApi`? If that's not available during build time, then the builds will indeed fail.
